Update GUI Skinning Code and warn about custom theme (#3825)
godotengine/godot-docs: tutorials/gui/gui_skinning.rst

used. This system allows for flexible overriding of themes in complex
user interfaces.

> .. attention::
>    Do not use the custom theme option in the project settings, as there
>    are bugs with the theme propagation. Instead apply your theme to the
>    root control!

Reviewer: ```suggestion
.. attention::

    Don't use the custom theme option in the Project Settings, as there
    are known bugs with theme propagation. Instead, apply your theme to the
    root Control node's Theme property. It will propagate to instanced scenes
    automatically. To get correct theming in the editor for instanced scenes,
    you can apply the theme resource to the instanced scene's root node as well.
```

.. tabs::
 .. code-tab:: gdscript GDScript

    var theme = Theme.new()
    ttheme.set_color("font_color", "Label", Color.red)

    var label = Label.new()
    label.theme = theme

 .. code-tab:: csharp

    var theme = new Theme();
    theme.SetColor("fontColor", "Label", new Color(1.0f, 0.0f, 0.0f));

    var label = new Label();
    label.Theme = theme;

In the example above, a new theme is created. The "font_color" option

> is changed and then applied to a label. Therefore the labels text (and all

Reviewer: ```suggestion
is changed and then applied to a label. Therefore, the label's text (and all
```

.. tabs::
 .. code-tab:: gdscript GDScript

    var label = Label.new()
    label.add_color_override("font_color", Color.red)

 .. code-tab:: csharp

    var label = new Label();
    label.AddColorOverride("fontColor", new Color(1.0f, 0.0f, 0.0f));

> In the inline help of Godot (in the script tab) you can check which theme options

Reviewer: ```suggestion
In the inline help of Godot (in the Script tab), you can check which theme options
```

From ``Styles``, open the "Normal" drop-down menu next to where it probably
says "null" and create a "New StyleBoxTexture", then
edit it. A texture stylebox contains a texture and the size of the margins
that will not stretch when the texture is stretched.

> This is called "3x3" stretching:

Reviewer: ```suggestion
This is called nine-patch or "3x3" stretching:
```
